tools/app/app/components/ContentArea.tsx

```diff
@@ -211,17 +235,27 @@ export const ContentArea: React.FC<ContentAreaProps> = ({
   const handleScroll = () => {
     const headers = document.querySelectorAll('.doc h1, .doc h2, .doc h3');
     const visibleHeaderIds: string[] = [];
-    headers.forEach((header) => {
+    let lastTitle: string | null = null;
+
+    const headerArray = Array.from(headers).reverse();
+
+    const boxRect = boxRef.current?.getBoundingClientRect();
+
+    for (const header of headerArray) {
       const rect = header.getBoundingClientRect();
-      if (
-        rect.top >= 0 &&
-        rect.bottom <= window.innerHeight &&
-        header.id &&
-        header.id !== ''
-      ) {
-        visibleHeaderIds.push(header.id);
+      if (rect.top < (boxRect?.top || -4) + 1) {
```

Comment:
Some weird magic...

Reply:
I guess that one needs a comment :smile: